registry/src/config.rs

```diff
@@ -14,12 +14,15 @@ pub enum RegistryValueData {
     QWord(u64),
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, JsonSchema)]
+#[derive(Debug, Default, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, JsonSchema)]
 #[serde(rename = "Registry", deny_unknown_fields)]
 pub struct Registry {
     /// The path to the registry key.
     #[serde(rename = "keyPath")]
     pub key_path: String,
+    /// The information from a config set --what-if operation.
+    #[serde(flatten, skip_serializing_if = "Option::is_none")]
+    pub what_if: Option<WhatIf>,
```

Comment:
I think for now, you should return this as:

```json
{
  "_metadata": {
    "whatIf": ["array of strings"]
  }
}
```

Then a follow-up PR to have the engine move `_metadata` member to the resource result `metadata`